tests/unit/test_authz.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+from unittest.mock import patch, MagicMock
+from time import sleep
+import flask
+from sheepdog.auth import authorize, AUTHZ_CACHE
+
+
+@pytest.fixture
+def mock_flask_app():
+    """Fixture to provide a test Flask app context."""
+    app = flask.Flask(__name__)
+    with app.app_context():
+        yield app
+
+
+@pytest.fixture
+def mock_auth_request():
+    """Fixture to mock the auth_request method."""
+    return MagicMock(return_value=True)
+
+
+@patch("sheepdog.auth.get_jwt_from_header", return_value="jwt")
+def test_authorize_caching(mock_jwt, mock_flask_app, mock_auth_request):
+    """
+    Unit test for `authorize` function to verify caching behavior.
+    Ensures that:
+    - The first call with a specific set of parameters invokes `auth_request`.
+    - Subsequent calls with the same parameters retrieve the response from cache.
+    - A new set of parameters triggers a new `auth_request` call.
+    """
+
+    AUTHZ_CACHE.clear()
+
+    with patch.object(flask, "current_app") as mock_app:
+        mock_app.auth.auth_request = mock_auth_request
```

Comment:
do you need `mock_flask_app` in addition to this?

Reply:
Yes, I tried without it but received errors related to missing flask context